src/game/overlay.c

```diff
@@ -78,14 +79,105 @@ static RGB888 m_ColorBarMap[][COLOR_STEPS] = {
       { 120, 60, 70 } },
 };
 
-static void Overlay_GetBarLocation(
-    int8_t bar_location, int32_t width, int32_t height, int32_t *x, int32_t *y);
-static void Overlay_DrawBar(int32_t value, int32_t value_max, int32_t bar_type);
+typedef struct BAR_INFO {
+    BAR_TYPE type;
+    int32_t value;
+    int32_t max_value;
+    int32_t percent;
```

Comment:
`percent` duplicates what can be calculated from `value` and `max_value`, I'd rather we'd store only one source of truth – in the current form it is unclear which is more valid during calculations.